libvast/vast/detail/notifying_stream_manager.hpp

```diff
@@ -19,7 +19,12 @@ namespace vast::detail {
 
 template <class State>
 void notify_listeners_if_clean(State& st, const caf::stream_manager& mgr) {
-  if (!st.flush_listeners.empty() && mgr.inbound_paths_idle()
+  // We intentionally don't check the inbound path state here because it will
+  // only be marked as idle after an ack was sent for the last batch that was
+  // received. However, acks are only sent once for each credit round, which
+  // means that sometimes we wouldn't notify even though all batches are done.
```

Comment:
I'm still wondering, in this scenario even if we "miss" the ack for the inbound path, wouldn't we notify the listeners after we received the ack from downstream for the forwarded packet?

Reply:
No, because sometimes `handle()` just isn't called any more. You can check my claim with this code.
```
  if (!st.flush_listeners.empty() && mgr.out().clean()) {
    VAST_INFO("{} inbound_idle = {}", st.self->name(), mgr.inbound_paths_idle());
    const auto& xs = mgr.inbound_paths();
    for (const auto& x : xs) {
      VAST_INFO("{} last_bid = {}; last_acked = {}; assigned_credit = {}",
                st.self->name(), x->last_batch_id, x->last_acked_batch_id, x->assigned_credit);
    }
    if (mgr.inbound_paths_idle())
      st.notify_flush_listeners();
```

Run with
```fish
while :; zcat vast/integration/data/zeek/conn.log.gz | ~/t/build/vast/detemp/bin/vast -v import -b --batch
-size=1 -n 242 zeek
end
```